ValveKeyValue/ValveKeyValue/Deserialization/KVObjectBuilder.cs

```diff
@@ -10,14 +10,14 @@ class KVObjectBuilder : IParsingVisitationListener
 
         public KVObject GetObject()
         {
-            if (stateStack.Count != 1)
+			foreach (var associatedBuilder in associatedBuilders)
```

Comment:
Why is the order of these operations swapped?

Reply:
That are swapped to handle the case where a file consists only of `#base` or `#include` directives. If they are not swapped, an error is thrown because the root file has no root object, and so `stateStack.Count` is `0`. By swapping these operations, the `#base` and `#include` directives are processed first, which means they have a chance to append their content and therefore make a valid root object.